pkg/systemd/notifyproxy/notifyproxy.go

```diff
@@ -100,34 +114,69 @@ func (p *NotifyProxy) waitForReady() {
 	go func() {
 		// Read until the `READY` message is received or the connection
 		// is closed.
-		const bufferSize = 1024
+
+		fdSize := unix.CmsgSpace(4)
 		sBuilder := strings.Builder{}
 		for {
-			for {
-				buffer := make([]byte, bufferSize)
-				num, err := p.connection.Read(buffer)
-				if err != nil {
-					if !errors.Is(err, io.EOF) {
-						p.errorChan <- err
-						return
-					}
-				}
-				sBuilder.Write(buffer[:num])
-				if num != bufferSize || buffer[num-1] == '\n' {
-					// Break as we read an entire line that
-					// we can inspect for the `READY`
-					// message.
-					break
+			buffer := make([]byte, _notifyBufferMax)
+			oob := make([]byte, _notifyFdMax*fdSize)
```

Comment:
Maybe you should allocate the buffers once, outside the loop?

Reply:
That's a nice idea, thanks for reviewing!